relayer/src/chain/cosmos/compatibility.rs

```diff
@@ -72,51 +92,50 @@ pub enum Diagnostic {
 /// Sdk module by name, as well as the constant
 /// [`SDK_MODULE_VERSION_REQ`] for version compatibility
 /// requirements.
-pub(crate) fn run_diagnostic(v: VersionInfo) -> Option<Diagnostic> {
-    let app_info = AppInfo {
-        app_name: v.app_name,
-        version: v.version,
-        git_commit: v.git_commit,
-    };
-
+pub(crate) fn run_diagnostic(version_info: VersionInfo) -> Result<(), Diagnostic> {
     // Parse the requirements into a semver
     let reqs = semver::VersionReq::parse(SDK_MODULE_VERSION_REQ)
         .expect("parsing the SDK module requirements into semver");
 
-    // Find the Cosmos SDK module
-    match v
+    // Get the Cosmos SDK version
+    let mut version = get_sdk_version(&version_info)?;
+
+    // Remove the pre-release version to ensure we treat pre-releases of the SDK
+    // as their normal version, eg. 0.42.0-rc2 should satisfy >=0.41.3, <= 0.42.6.
+    version.pre = semver::Prerelease::EMPTY;
+
+    // Finally, check the version requirements
+    match reqs.matches(&version) {
+        true => Ok(()),
+        false => Err(Diagnostic::MismatchingSdkModuleVersion {
+            requirements: SDK_MODULE_VERSION_REQ.to_string(),
+            found: version.to_string(),
+            app: AppInfo::from(version_info),
+        }),
+    }
+}
+
+fn get_sdk_version(version_info: &VersionInfo) -> Result<semver::Version, Diagnostic> {
+    let module = version_info
         .build_deps
         .iter()
         .find(|&m| m.path.contains(SDK_MODULE_NAME))
-    {
-        None => Some(Diagnostic::SdkModuleNotFound {
+        .ok_or_else(|| Diagnostic::SdkModuleNotFound {
```

Comment:
Just a question (unrelated to this PR) - I was curious as to why we use the `VersionInfo::build_deps::<Module>::version` instead of the `VersionInfo::cosmos_sdk_version` to check for compatibility?